README.md

```diff
@@ -291,6 +291,38 @@ echo 'a';
 
 See also [example #3](examples).
 
+### addSignal()
+
+**Note: Signaling is only available on Unix-like platform, Windows isn't supported.**
+
+The `addSignal(int $signal, callable $listener): void` method can be used to
+be notified about OS signals. This is useful to catch user interrupt signals or 
+shutdown signals from tools like `supervisor` or `systemd`.
+
+```php
+$listener = function (int $signal) {
+    echo 'Caught user iterrupt signal', PHP_EOL;
+};
+$loop->addSignal(SIGINT, $listener);
+```
+
+See also [example #4](examples).
+
+**Note: A listener can be added more then once to the same signal and it will be called 
+for each time it is added. But on [`removeSignal`](#removeSignal) it is removed for all 
+the times it has been added. 
```

Comment:
Removing a single instance would be easy to do. But it might be more consistent to only allow one occurrence of a instance per signal in the first place.